cub/cub/device/dispatch/dispatch_streaming_reduce.cuh

```diff
@@ -0,0 +1,451 @@
+ * if the user uses a different index type than the global offset type used by the algorithm.
+ */
+template <typename KeyValuePairOutItT>
+struct write_to_user_out_it
+{
+  KeyValuePairOutItT out_it;
+
+  template <typename IndexT, typename OffsetT, typename ResultT>
+  _CCCL_DEVICE _CCCL_FORCEINLINE void operator()(IndexT, cub::KeyValuePair<OffsetT, ResultT> reduced_result)
+  {
+    using kv_pair_t = ::cuda::std::_If<
+      ::cuda::std::is_assignable<decltype(*out_it), cub::KeyValuePair<::cuda::std::uint64_t, ResultT>>::value,
+      cub::KeyValuePair<::cuda::std::uint64_t, ResultT>,
+      ::cuda::std::_If<
+        ::cuda::std::is_assignable<decltype(*out_it), cub::KeyValuePair<::cuda::std::int64_t, ResultT>>::value,
+        cub::KeyValuePair<::cuda::std::int64_t, ResultT>,
+        ::cuda::std::_If<
+          ::cuda::std::is_assignable<decltype(*out_it), cub::KeyValuePair<::cuda::std::uint32_t, ResultT>>::value,
+          cub::KeyValuePair<::cuda::std::uint32_t, ResultT>,
+          cub::KeyValuePair<::cuda::std::int32_t, ResultT>>>>;
+
+    using index_t = typename kv_pair_t::Key;
+    *out_it       = kv_pair_t{static_cast<index_t>(reduced_result.key), reduced_result.value};
+  }
+};
+
+/******************************************************************************
+ * Single-problem streaming reduction dispatch
+ *****************************************************************************/
+
+//! @rst
+//! Utility class for dispatching a streaming device-wide argument extremum computation, like `ArgMin` and `ArgMax`.
+//! Streaming, here, refers to the approach used for large number of items that are processed in multiple partitions.
+//!
+//! @tparam InputIteratorT
+//!   Random-access input iterator type for reading input items @iterator
+//!
+//! @tparam OutputIteratorT
+//!   Output iterator type for writing the result of the (index, extremum)-key-value-pair
+//!
+//! @tparam PerPartitionOffsetT
+//!   Offset type used as the index to access items within one partition, i.e., the offset type used within the kernel
+//! template specialization
+//!
+//! @tparam GlobalOffsetT
+//!   Offset type used as the index to access items within the total input range, i.e., in the range [d_in, d_in +
+//! num_items)
+//!
+//! @tparam ReductionOpT
+//!   Binary reduction functor type having member
+//!   `auto operator()(const T &a, const U &b)`
+//!
+//! @tparam InitT
+//!   Initial value type
+//!
+//! @tparam PolicyChainT
+//!   The policy chain passed to the DispatchReduce template specialization
+//! @endrst
+template <typename InputIteratorT,
+          typename OutputIteratorT,
+          typename PerPartitionOffsetT,
+          typename GlobalOffsetT,
+          typename ReductionOpT,
+          typename InitT,
+          typename PolicyChainT =
+            DeviceReducePolicy<KeyValuePair<PerPartitionOffsetT, InitT>, PerPartitionOffsetT, ReductionOpT>>
+struct DispatchStreamingArgReduce
```

Comment:
Suggestion: For recently added dispatchers, we aimed to make them look like internal types, so users are less likely to reach out to those types. So we applied the usual naming scheme `dispatch_streaming_arg_reduce_t`. Also, since it's an internal type, I would also not publicly document the type. You can leave your documentation, but avoid doxygen comments.

E.g., for `cub::DeviceTransform` the dispatcher is called `cub::detail::transform::dispatch_t`.